client/types/authorizations.d.ts

```diff
@@ -27,3 +28,19 @@ export interface AuthorizationsSummary {
 	store_currencies?: string[];
 	customer_currencies?: string[];
 }
+
+export interface AuthorizationsState {
+	summary: {
+		[ x: string ]: { data?: AuthorizationsSummary; error?: string };
+	};
+	byId: Record< string, Authorization >;
+}
+
+interface AuthorizationsList {
+	data: Authorization[];
```

Comment:
If only we didn't have the `data` field, we could get rid of this type in favor of `Authorization[]`.

Reply:
Thanks @anu-rock that's a great point. I admit I introduced this type because I was struggling to type the `getAuthorizations` generator in `resolvers`. Since generator function is typed as `unknown`, TS was complaining when I accessed `result.data` I forgot that later I found a different approach that I liked better, but forgot to apply here. 

I believe we were missing a type for the response of the endpoint, which I added in this commit [40b7dc3](https://github.com/Automattic/woocommerce-payments/pull/4961/commits/40b7dc33abd12aa53541be3aceccf936d6ee3ae9). Now TS is happy, and so am I.

Reply:
That looks like a good middle ground to me :)